### PR TITLE
feat: historical price query [SF-1362]

### PR DESCRIPTION
**Pull request by vijayeth**

### CODE REVIEW

**Review by biga816 — COMMENTED** [review]
Why this queries get only 1 record pre query?
Also, it looks the query name and actual query doesn't match.

**Comment by vijayeth** [comment]
> Why this queries get only 1 record pre query? Also, it looks the query name and actual query doesn't match.

@biga816  It’s ordered by `createdAt desc` and using `first: 1` will give the most recent transaction that happened just before or at that interval for particular maturity.
Also, I'll name the query as `HistoricalPriceQuery`.

**Comment by biga816** [comment]
@vijayeth  Why you need to get the most recent transaction per interval time?
It's not a historical price. It looks like just last transactions for each interval.

**Comment by vijayeth** [comment]
> @vijayeth Why you need to get the most recent transaction per interval time? It's not a historical price. It looks like just last transactions for each interval.

@biga816 The transaction gives the last execution price for that interval & maturity — which we're using to calculate the Rate.

**Comment by biga816** [comment]
@vijayeth  Then it's not a query to get a historical price. Could you change the query name?

File: packages/sf-graph-client/src/queries/development/lending-market.ts [hunk 1]
```diff
@@ -294,3 +294,49 @@ export const TRANSACTION_CANDLE_STICK = gql`
...
+    maturityList: number[],
+    currency: string
+) => {
+    if (!intervals.length || !maturityList.length || !currency) {
+        return gql`
+            query FallBackQuery {
+                __typename
+            }
+        `;
+    }
```
Comment (biga816) [inline packages/sf-graph-client/src/queries/development/lending-market.ts]: What does this need for?

Comment (vijayeth) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: @biga816 Had to add this fallback because in tests, the dynamic query was null or undefined for different scenarios , causing Apollo to throw error, as it needs a valid query.

Comment (vijayeth) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: Also , would you please suggest the name for query out of these -
`GET_EXECUTION_PRICE` or `GET_TRANSACTIONS_FOR_HISTORICAL_YIELD_GRAPH`

File: packages/sf-graph-client/src/queries/development/lending-market.ts [hunk 2]
```diff
@@ -294,3 +294,49 @@ export const TRANSACTION_CANDLE_STICK = gql`
...
+    const queryParts = intervals
+        .map((interval, i) => {
+            return maturityList.map((maturity, j) => {
+                return `
+        tx${i}_${j}: transactions(
+          where: {
+            createdAt_lte: ${interval}
```
Comment (biga816) [inline packages/sf-graph-client/src/queries/development/lending-market.ts]: Does `interval` has value in date type?
I think generally `interval` has fixed time like 10s, 60min, or something like that.
So if this name is changed to something related to date, it can be easier to understand what this is.

Comment (vijayeth) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: Interval is a time offset in seconds, used as `currentTime - timestamp`, renaming the query to `TRANSACTIONS_BY_TIMESTAMP_AND_MATURITY_QUERY` & intervals to timestamp.

Comment (vijayeth) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: renamed in c1d462935e9fd9a2e851d12fbac7e569fa0600cf

File: packages/sf-graph-client/src/queries/development/lending-market.ts (hunk 1 above)
Comment (biga816) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: Does this query return empty?
If yes, you can't notice if your inputs are wrong or there is no transactions.

Comment (vijayeth) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: it only returns empty when any of the parameters is missing.

Comment (biga816) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: Then you can't notice if inputed parameters are wrong or there is no transactions when you got empty.
How about put the code to throw an error there? Is there any issue with this solution?

Comment (vijayeth) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: I did try throwing an error, but it broke the app since Apollo always expects a valid GraphQL query. In some cases (especially in tests), variables like intervals or maturityList aren’t ready yet, so I added this fallback. we can skip tests for now and remove this.

Comment (vijayeth) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: I did try throwing an error, but it broke the app since Apollo always expects a valid GraphQL query. In some cases (especially in tests), variables like intervals or maturityList aren’t available which breaks the tests, so I added this fallback. 
- if any of the parameters is empty, we get empty query.
- if inputed parameters are wrong or there is no transaction then query return empty transaction array for each timestamp & maturity combination , so in sf-app we mark those aprs as `0`.

Comment (biga816) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: >  I did try throwing an error, but it broke the app

What's the issue of this?
If the parameters are wrong, it means the SDK user set the wrong parameters in the code.
The code that call this SDK must always set the correct parameters.

Comment (vijayeth) [reply packages/sf-graph-client/src/queries/development/lending-market.ts]: @biga816 I've thrown errors if wrong parameters have been passed. There were issues with test cases at FE, I skipped them earlier and made fixes too.
fixed in cd5a47a5d38956d8bf889d61f1f938e4209421a0

**Review by biga816 — APPROVED** [review]
